string_set.go

```diff
@@ -0,0 +1,33 @@
+package main
+
+import (
+	"strings"
+)
+
+type StringSet []string
+
+func NewStringSet(s string) StringSet {
+	if len(s) == 0 {
+		return []string{}
+	} else {
```

Comment:
Drop the `else`.

```go
if specialCase {
  return ...
}
return ...
```

Where in Ruby the entire `if` would be an expression that yields a result -- and you would tend to favour this form -- in Go, special cases tend to be emphasised using indented statements (`if`, `case`, etc.) with the 'most typical case' reading down the page without any indentation.